receiver/splunksearchapireceiver/receiver.go

```diff
@@ -0,0 +1,320 @@
+	if err != nil {
+		return fmt.Errorf("failed to get storage client: %w", err)
+	}
+	ssapir.storageClient = storageClient
+
+	err = ssapir.initCheckpoint(cancelCtx)
+	if err != nil {
+		return fmt.Errorf("failed to initialize checkpoint: %w", err)
+	}
+	go ssapir.runQueries(cancelCtx)
+	return nil
+}
+
+func (ssapir *splunksearchapireceiver) Shutdown(ctx context.Context) error {
+	ssapir.logger.Debug("shutting down logs receiver")
+	if ssapir.cancel != nil {
+		ssapir.cancel()
+	}
+
+	err := ssapir.checkpoint(ctx)
+	if err != nil {
+		ssapir.logger.Error("failed checkpoint", zap.Error(err))
+	}
+
+	return ssapir.storageClient.Close(ctx)
+}
+
+func (ssapir *splunksearchapireceiver) runQueries(ctx context.Context) error {
+	for _, search := range ssapir.config.Searches {
+		// set current search query
+		ssapir.checkpointRecord.Search = search.Query
+
+		// set default event batch size (matches Splunk API default)
+		if search.EventBatchSize == 0 {
+			search.EventBatchSize = 100
+		}
+
+		// create search in Splunk
+		searchID, err := ssapir.createSplunkSearch(search)
+		if err != nil {
+			ssapir.logger.Error("error creating search", zap.Error(err))
+			return err
+		}
+
+		// wait for search to complete
+		if err = ssapir.pollSearchCompletion(ctx, searchID); err != nil {
+			ssapir.logger.Error("error polling for search completion", zap.Error(err))
+			return err
+		}
+
+		for {
+			ssapir.logger.Info("fetching search results")
+			results, err := ssapir.getSplunkSearchResults(searchID, offset, search.EventBatchSize)
+			if err != nil {
+				ssapir.logger.Error("error fetching search results", zap.Error(err))
+				return err
+			}
+			ssapir.logger.Info("search results fetched", zap.Int("num_results", len(results.Results)))
+
+			// parse time strings to time.Time
+			earliestTime, err := time.Parse(time.RFC3339, search.EarliestTime)
+			if err != nil {
+				ssapir.logger.Error("earliest_time failed to be parsed as RFC3339", zap.Error(err))
+			}
+
+			latestTime, err := time.Parse(time.RFC3339, search.LatestTime)
+			if err != nil {
+				ssapir.logger.Error("latest_time failed to be parsed as RFC3339", zap.Error(err))
+			}
+
+			logs := plog.NewLogs()
+			for idx, splunkLog := range results.Results {
+				if (idx+exportedEvents) >= search.Limit && search.Limit != 0 {
+					limitReached = true
+					break
+				}
+				// convert log timestamp to ISO8601 (UTC() makes RFC3339 into ISO8601)
+				logTimestamp, err := time.Parse(time.RFC3339, splunkLog.Time)
+				if err != nil {
+					ssapir.logger.Error("error parsing log timestamp", zap.Error(err))
+					break
+				}
+				if logTimestamp.UTC().After(latestTime.UTC()) {
+					ssapir.logger.Info("skipping log entry - timestamp after latestTime", zap.Time("time", logTimestamp.UTC()), zap.Time("latestTime", latestTime.UTC()))
+					// logger will only log up to 10 times for a given code block, known weird behavior
```

Comment:
Is this a fairly common occurrence? If so it may warrant downgrading to a debug log?

Reply:
It was common before I sent the time params to Splunk, but these are technically a little redundant now. They should only be getting hit if something goes wonky with the time zones.

Reply:
Sounds good, is there an issue you can maybe link to here? It might be okay to remove this comment since you said it doesn't really happen anymore, but at the same time if it does happen again this comment would be helpful to have along with a link to the underlying issue/explanation. I'll leave it up to you.